### PR TITLE
Adding LCM1602A1 bindings

### LINKED ISSUE #171: Add bindings for LCM1602C / hd44780  (LCD Char display)
cc: @joperezr 

Not sure if this will be generic hd44780 or if there is something specific to LCM1602C

### PR DESCRIPTION
**Pull request by joperezr**
This will add the LCM1602A1 binding.

It has been tested on an LCM1602C and on a CrowPi as well.

cc: @shaggygi @krwq 

Fixes #171

### CODE REVIEW

**Comment by shaggygi** [comment]
@JeremyKuhne  too late #125 (if this is what you meant)  :smile:

**Comment by JeremyKuhne** [comment]
@shaggygi Thanks- I'll take a look!

**Comment by JeremyKuhne** [comment]
Should this be called an HD44780 binding?

**Comment by joperezr** [comment]
> Should this be called an HD44780 binding?

The problem with that is that we do have some very specific initialization for this display in the code:

https://github.com/dotnet/iot/blob/583948132b3c720e0ac1451dc4f5261e1883b9a4/src/devices/Lcm1602a1/Lcm1602a1.cs#L260-L266

Because of that, what I think is a better idea is to add an abstract Hd44780 class that defines all of the shared logic, and then just create subclasses that define the device-specific setup. If we are going with that approach, I would keep this code as is for now, and then once we add a second display perform the refactoring.

**Comment by joperezr** [comment]
@shaggygi let me know what you think about the changes I made on the Mcp device

File: src/devices/Lcm1602a1/Lcm1602a1.cs [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+                SetPinMode(_dataPins[i], PinMode.Output);
+            }
+
+            // SEE PAGE 45/46 FOR INITIALIZATION SPECIFICATION!
+            // according to datasheet, we need at least 40ms after power rises above 2.7V
+            // before sending commands. Arduino can turn on way before 4.5V so we'll wait 50
+            DelayMicroseconds(50000);
+            
+            // Initialize the display.
+            Write((byte)0x33);
```
Comment (JeremyKuhne) [inline src/devices/Lcm1602a1/Lcm1602a1.cs]: I'm having a hard time tying this back to the datasheet. It looks like we were trying to reset the display if I'm following page 45 correctly. Is this necessary? If so, it looks like you need to send 00110000 three times with waits after the first two (4.1ms and 100μs) (without checking BF).  This appears to me like we're not really resetting?

On page 27 it says function set instruction must be run before all other instructions. On page 40 it shows function set before display control and entry mode.

I went back and checked the samples again on the CrowPi and I think I'm seeing behavior that backs this up. If I start up from power off the code doesn't work and prints garbage. If I then run the CrowPi python sample the display gets reset correctly and the sample works from that point. When I tested before I didn't power off the CrowPi.

**Comment by joperezr** [comment]
Closing this in favor of #206 which is adding a more complete version of the same binding.

**Pull request closed without merging**